consent/handler.go

```diff
@@ -573,4 +575,37 @@ func (h *Handler) RejectConsentRequest(w http.ResponseWriter, r *http.Request, p
 	h.H.Write(w, r, &RequestHandlerResponse{
 		RedirectTo: urlx.SetQuery(ru, url.Values{"consent_verifier": {request.Verifier}}).String(),
 	})
+}
+	})
+}
+
+// swagger:route DELETE /oauth2/auth/logout oAuth2 logoutUser
+//
+// Logs user out by deleting the session cookie
+//
+// This endpoint deletes ths user's login session cookie and redirects the browser to the url
+// listed in `LOGOUT_REDIRECT_URL` environment variable.
+//
+//
+//     Consumes:
+//     - text/html
+//
+//     Produces:
+//     - text/html
+//
+//     Schemes: http, https
+//
+//     Responses:
+//       302: emptyResponse
+//       404: genericError
+//       500: genericError
+
+func (h *Handler) LogoutUser(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	err := revokeAuthenticationSession(w, r)
+	if err != nil {
+		h.H.WriteError(w, r, err)
+		return
+	}
+
+	http.Redirect(w, r, os.Getenv("LOGOUT_REDIRECT_URL"), 302)
```

Comment:
This solution is ok, but it makes setting defaults and testing the configuration harder. Typically what we do here is:

1. Add a config item in [config/config.go](https://github.com/ory/hydra/blob/master/config/config.go#L69)
2. Add a field (e.g. `LogoutRedirectURL`) to set this in the [handler.go](https://github.com/ory/hydra/blob/master/consent/handler.go#L41) and the respective factors (`new...`)
3. Bind and set a default in `cmd/root.go` like [here](https://github.com/ory/hydra/blob/master/cmd/root.go#L119-L120).
4. Use `LogoutRedirectURL` here instead of `os.GetEnv()`

Additionally, we should set up a default handler for the log out screen, similar to [this one](https://github.com/ory/hydra/blob/master/oauth2/handler_fallback_endpoints.go#L52) and add it to the frontend router [here](https://github.com/ory/hydra/blob/master/oauth2/handler.go#L161). The layout of the page doesn't matter. It should read something "You have been logged out successfully. ... Admin forgot to set up the redirect URL after logout ... to fix that set environment variable `LogoutRedirectURL`, e.g. `export LogoutRedirectURL=http://...` or on windows `set LogoutRedirectURL=http://...` ..."